core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,277 @@
+            interactionId));
+      } else {
+        $scope.objectType = null;
+      }
+    });
+
+    $scope.getSolutionSummary = function() {
+      var solution = stateSolutionService.displayed;
+      var isExclusiveAnswer = (
+        solution.answerIsExclusive ? 'Only' : 'One');
+      var correctAnswer = StateSolutionHelperService.getCorrectAnswer(solution);
+      var explanation = (
+        $filter('convertToPlainText')(solution.explanation));
+      return (
+        '[' + isExclusiveAnswer + ' solution is ' + correctAnswer + '] ' +
+        explanation);
+    };
+
+    // This returns false if the current interaction ID is null.
+    $scope.isCurrentInteractionLinear = function() {
+      var interactionId = stateInteractionIdService.savedMemento;
+      return interactionId && INTERACTION_SPECS[interactionId].is_linear;
+    };
+
+    $scope.openAddSolutionModal = function() {
+      alertsService.clearWarnings();
+      $rootScope.$broadcast('externalSave');
+      $scope.isActive = false;
+
+      $modal.open({
+        templateUrl: 'modals/addSolution',
+        backdrop: 'static',
+        scope: $scope,
+        controller: [
+          '$scope', '$rootScope', '$modalInstance', 'editorContextService',
+          function($scope, $rootScope, $modalInstance, editorContextService) {
+            $scope.EXPLANATION_FORM_SCHEMA = {
+              type: 'html',
+              ui_config: {}
+            };
+
+            $scope.INTERACTION_SPECS = INTERACTION_SPECS;
+
+            $scope.tmpSolution = {};
+            $scope.tmpSolution.answerIsExclusive = false;
+            $scope.tmpSolution.correctAnswer = '';
+            $scope.tmpSolution.explanation = '';
+
+            $scope.addSolutionForm = {};
+
+            if ($scope.objectType === 'CodeString') {
+              stateSolutionService.displayed = {
+                correctAnswer: {
+                  code: '',
+                  output: '',
+                  evaluation: '',
+                  error: ''
+                }
+              };
+            }
+
+            $scope.saveSolution = function() {
+              // Close the modal and save it afterwards.
+              var answer = ($scope.correctAnswer !== null) ? (
+                $scope.correctAnswer) : $scope.tmpSolution.correctAnswer;
+              if ($scope.objectType === 'CodeString') {
+                answer = {
+                  code: $scope.tmpSolution.correctAnswer,
+                  output: '',
+                  evaluation: '',
+                  error: ''
+                };
+              } else if ($scope.objectType === 'UnicodeString') {
+                answer = {
+                  ascii: '',
+                  latex: $scope.tmpSolution.correctAnswer
+                };
+              }
+
+              $modalInstance.close({
+                solution: angular.copy(
+                  SolutionObjectFactory.createNew(
```

Comment:
Given that you're creating a new solution do you really need the angular.copy()? I'm fine either way but just wanted to check.